### PR TITLE
Add workload identity federation docs

### PR DESCRIPTION
**Pull request by asatwal**
### Context
Add workload identity federation documentation to allow teams to setup on their own projects.

### Trello tickets:
https://trello.com/c/F7WrGYYq

### Changes proposed in this pull request
Documentation only with full WIF instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,33 +99,60 @@ This will create some configuration files, including `config/initializers/dfe_an
...
-Depending on how your app environments are set up, we recommend you use a
-service account created for the `development` environment to test your local
-machine's integration with BigQuery. This requires that your project is set up
-in Google Cloud as per [the instructions above](docs/google_cloud_bigquery_setup.md).
+Ensure project is set up as per the [Google Cloud setup guide](docs/google_cloud_bigquery_setup.md) above.
 
-1. Access the `development` service account you previously set up
-1. Go to the keys tab, click on "Add key" > "Create new key"
-1. Create a JSON private key. This file will be downloaded to your local system.
+Set the following environment variables for your Rails app.
```
Comment (ericaporter) [inline README.md:107]: Maybe `.....Rails app:`

File: README.md [hunk 2]
```diff
@@ -99,33 +99,60 @@ This will create some configuration files, including `config/initializers/dfe_an
...
+BIGQUERY_DATASET=your-bigquery-dataset-name
+```
+
+If you're stuck with differently-named env vars, you can configure the names in
+`dfe_analytics.rb`.
+
+### 3. BigQuery authentication method
+
+### 3.1 Workload Identity Federation
+
```
Comment (ericaporter) [inline README.md:120]: Maybe #### for 3.1

File: docs/google_cloud_bigquery_setup.md [hunk 3]
```diff
@@ -501,14 +501,81 @@ in the new dataset.
...
+
+With Workload Identity Federation (WIF), you can use Identity and Access Management (IAM) to grant to external identities [IAM roles](https://cloud.google.com/iam/docs/overview#roles), direct access on Google Cloud resources. You can also grant access through service account impersonation.
+
+With DfE::Analytics our strong preference is to use WIF where possible. Where WIF is not possible to use then OAuth should be considered. The use of service account API Keys is discouraged.
+
+The diagram below demonstrates our use of WIF within DfE Analytics connecting from an Azure client to BigQuery.
+
+![[azure-gcp-wif.svg]]
+
+
```
Comment (ericaporter) [inline docs/google_cloud_bigquery_setup.md:529]: Extra line (unless this follows formatting)

File: docs/google_cloud_bigquery_setup.md [hunk 4]
```diff
@@ -501,14 +501,81 @@ in the new dataset.
...
+If this does not exist then follow the steps above:
+- [Create an appender service account](#3-create-an-appender-service-account)
+- [Give the service account access to your dataset](#3-create-an-appender-service-account)
+
+### 5. Service account permissions for workload identity federation
+
+The service account defined in step 4 above should be granted access using service account impersonation.
+
+If this does not exist then access can be granted with either the [update wif service account permissions ](https://github.com/DFE-Digital/teacher-services-analytics-cloud/blob/main/scripts/gcloud/update-wif-service-account-permissions.sh) gcloud script or from the [IAM](https://console.cloud.google.com/iam-admin/workload-identity-pools/pool/azure-cip-identity-pool) gcloud console, by navigating to the "GRANT ACCESS" window. Use the attributes specified in the gcloud script. Note that the subject must be set to the Managed Identity Object ID from azure for each environment (see Step 1 above).
```
Comment (ericaporter) [inline docs/google_cloud_bigquery_setup.md:573]: from 'Azure'

**Review by ericaporter — APPROVED** [review]
👍 Looks good

File: docs/google_cloud_bigquery_setup.md [hunk 5]
```diff
@@ -501,14 +501,78 @@ in the new dataset.
...
+
+### 6. Download the WIF client credentials
+
+Download the JSON WIF Credentials file and set to following environment variables to the content of this file:
+
+`GOOGLE_CLOUD_CREDENTIALS`
+
+Download the JSON WIF Credentials file either the [create wif client credentials](https://github.com/DFE-Digital/teacher-services-analytics-cloud/blob/main/scripts/gcloud/create-wif-client-credentials.sh) gcloud script or from the [IAM](https://console.cloud.google.com/iam-admin/workload-identity-pools/pool/azure-cip-identity-pool) gcloud console, by navigating to the "CONNECTED SERVICE ACCOUNTS" tab. Use the the attributes specified in the gcloud script.
```
Comment (stevenleggdfe) [inline docs/google_cloud_bigquery_setup.md:578]: Missing a word or two in this sentence I think

**Review by stevenleggdfe — CHANGES_REQUESTED** [review]

**Review by stevenleggdfe — APPROVED** [review]

**Pull request merged**